### PR TITLE
Allow RC and milestone releases

### PR DESCRIPTION
**Pull request by SimY4**
Allow to specify milestone and release candidate versions of scala standard library and compiler.

### CODE REVIEW

**Review by kbielefe — COMMENTED** [review]
:+1:  Would it make sense to consider making the `RC` and `M` case insensitive?

File: src/test/groovy/unit/TestScalaMultiVersionPlugin.groovy [hunk 1]
```diff
@@ -157,4 +171,21 @@ class TestScalaMultiVersionPlugin extends GroovyTestCase implements SimpleProjec
         assert root.dependencies.'*'.find { it.artifactId == "fake-scala-dep_2.12" } != null
     }
 
+    void testMavenPomMilestoneRelease() {
```
Comment (nathanalderson) [inline src/test/groovy/unit/TestScalaMultiVersionPlugin.groovy]: Can we integrate this test with the previous one?

```scala
void testMavenPom() {
    ["2.12.1", "2.13.0-M5"].each {
        ...
    }
}
```

**Review by nathanalderson — CHANGES_REQUESTED** [review]
Great changes!

One request inline. Also, you need to include a commit whose comment begins with either `feat:` for new feature or `fix:` for a fix (I think this PR could be considered either). This will make the semantic versioning tool bump the version number correctly. The easiest thing would be to just make sure your next commit includes `fix:`. You could also rewrite the first commit message (:see_no_evil:) or add an empty commit if you feel like it. I'm not picky.

**Comment by SimY4** [comment]
@kbielefe if I may, I would suggest making this regex configurable by the end user with a reasonable default (say, standard Major.Minor.Patch regex). Just so if scala community decide to follow a different versioning style, code-change will not be necessary.

**Review by nathanalderson — APPROVED** [review]

**Comment by nathanalderson** [comment]
Configurable is always nice. The convention for now seems to be capital letters though, and the maven resolution seems to be case-sensitive, so I think the current regex is fine as is or as a default if you want to make it configurable.

**Comment by SimY4** [comment]
@nathanalderson I can raise another PR to make it configurable.

Thanks

**Comment by SimY4** [comment]
Is there something else I should do to move forward with these changes?

**Comment by nathanalderson** [comment]
Sorry! I must have missed your previous comment about raising another PR. That works for me. Thanks for contributing!

**Pull request merged**